### PR TITLE
git-relevant-history: init package at 2022-09-15

### PR DESCRIPTION
**Pull request by bendlas**
###### Description of changes

Add [git-relevant-history](https://github.com/rainlabs-eu/git-relevant-history/) package.

This can separate git file histories, while tracking renames.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/version-management/git-relevant-history/default.nix [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  src = fetchFromGitHub {
+    owner = "rainlabs-eu";
+    repo = pname;
+    rev = "84552324d7cb4790db86282fc61bf98a05b7a4fd";
+    hash = "sha256-46a6TR1Hi3Lg2DTmOp1aV5Uhd4IukTojZkA3TVbTnRY=";
+  };
+  propagatedBuildInputs = [
+    git git-filter-repo
+    python3.pkgs.docopt
+  ];
```
Comment (yl3dy) [inline pkgs/applications/version-management/git-relevant-history/default.nix:15]: Please add metadata.

**Review by yl3dy — CHANGES_REQUESTED** [review]

File: pkgs/applications/version-management/git-relevant-history/default.nix (hunk 1 above)
Comment (bendlas) [reply pkgs/applications/version-management/git-relevant-history/default.nix:15]: good point, done

**Comment by yl3dy** [comment]
Result of `nixpkgs-review pr 238952` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>git-relevant-history</li>
    <li>git-relevant-history.dist</li>
  </ul>
</details>

**Review by yl3dy — APPROVED** [review]
LGTM!

**Pull request merged**